iam/api-client/src/main/java/iam/snippets/Quickstart.java

```diff
@@ -16,164 +16,159 @@
     // Gets the project's policy.
-    Policy policy = getPolicy(crmService, projectId);
+    Policy policy = getPolicy(iamClient, projectId, serviceAccount);
 
     // If policy is not retrieved, return early.
     if (policy == null) {
       return;
     }
 
+    Policy.Builder updatedPolicy = policy.toBuilder();
+
     // Finds binding in policy, if it exists.
     Binding binding = null;
-    for (Binding b : policy.getBindings()) {
+    for (Binding b : updatedPolicy.getBindingsList()) {
       if (b.getRole().equals(role)) {
-        binding = b; 
+        binding = b;
         break;
       }
     }
 
     if (binding != null) {
       // If binding already exists, adds member to binding.
-      binding.getMembers().add(member);
+      binding.getMembersList().add(member);
     } else {
       // If binding does not exist, adds binding to policy.
-      binding = new Binding();
-      binding.setRole(role);
-      binding.setMembers(Collections.singletonList(member));
-      policy.getBindings().add(binding);
+      binding = Binding.newBuilder()
+              .setRole(role)
+              .addAllMembers(Collections.singletonList(member))
+              .build();
+      updatedPolicy.addBindings(binding);
     }
 
     // Sets the updated policy.
-    setPolicy(crmService, projectId, policy);
+    setPolicy(iamClient, projectId, serviceAccount, updatedPolicy.build());
   }
 
   public static void removeMember(
-      CloudResourceManager crmService, String projectId, String member, String role) {
+          IAMClient iamClient, String projectId, String serviceAccount, String member, String role) {
     // Gets the project's policy.
-    Policy policy = getPolicy(crmService, projectId);
+    Policy.Builder policy = getPolicy(iamClient, projectId, serviceAccount).toBuilder();
 
     // Removes the member from the role.
     Binding binding = null;
-    for (Binding b : policy.getBindings()) {
+    for (Binding b : policy.getBindingsList()) {
       if (b.getRole().equals(role)) {
         binding = b;
         break;
       }
     }
-    if (binding != null && binding.getMembers().contains(member)) {
-      binding.getMembers().remove(member);
-      if (binding.getMembers().isEmpty()) {
-        policy.getBindings().remove(binding);
+
+    if (binding != null && binding.getMembersList().contains(member)) {
+      List<String> newMemberList = new ArrayList<>(binding.getMembersList());
+      newMemberList.remove(member);
+
+      Binding newBinding = binding.toBuilder().clearMembers()
+              .addAllMembers(newMemberList)
+              .build();
+      List<Binding> newBindingList = new ArrayList<>(policy.getBindingsList());
+      newBindingList.remove(binding);
+
+      if (!newBinding.getMembersList().isEmpty()) {
+        newBindingList.add(newBinding);
       }
+
+      policy.clearBindings()
+              .addAllBindings(newBindingList);
```

Comment:
Is there a simpler way of doing this instead of having to remove and add all bindings?

Reply:
all fields inside Policy are immutable Collections